### PR TITLE
Config bundle autowiring

### PR DESCRIPTION
**Pull request by escopecz**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [N]
| New feature/enhancement? (use the a.x branch)      | [Enhancement]
| Deprecations?                          | [Y]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [Not as it's a config change] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR removes all hard-coded services from config.php and use autowiring for them. Plus some type improvements.

The original service aliases are still configured, but marked as deprecated. This PR also replaces these aliases with FQCN wherever they were used.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open the Mautic global configuration
3. Change some value
4. Save the configuration
5. The value should remain saved

Basically, the configuration form must work as before with no errors.

### CODE REVIEW

**Review by RCheesley — APPROVED** [review]
Looks good, I went through and changed different types of settings in different bundles and everything works as I would expect it to.

Thanks @escopecz 🚀

**Comment by sdoering** [comment]
To me it looks good to, I went through nearly every setting, changed it, changed it back and everything works as I would expect it to.

Thanks @escopecz 👍
